tests/packagedcode/test_buildpack.py

```diff
@@ -0,0 +1,425 @@
+        expected_file = self.get_test_loc('buildpack/paketo-buildpacks/source-removal/expectedbuildpack.json')
+        result_file = self.get_temp_file('results.json')
+        run_scan_click(['--package', test_file, '--json-pp', result_file])
+        check_json_scan(expected_file, result_file, regen=REGEN_TEST_FIXTURES)
+
+    def test_parse_heroku_buildpack_java_toml(self):
+        test_file = self.get_test_loc('buildpack/heroku-buildpacks/heroku-buildpack-java/buildpack.toml')
+        result_packages = list(buildpack.BuildpackHandler.parse(test_file))
+        expected_packages = [
+            models.PackageData(
+                type=buildpack.BuildpackHandler.default_package_type,
+                datasource_id=buildpack.BuildpackHandler.datasource_id,
+                description="Heroku buildpack for Java",
+                name="Java",
+                version=None,
+                extra_data={
+                    "ignore_files": [
+                        "etc/",
+                        "spec/",
+                        "test/",
+                        ".gitignore",
+                        ".github/",
+                        "hatchet.json",
+                        "Gemfile",
+                        "Gemfile.lock"
+                    ]
+                }
+            )
+        ]
+        compare_package_results(expected_packages, result_packages)
+
+    def test_parse_heroku_buildpack_php_toml(self):
+        test_file = self.get_test_loc('buildpack/heroku-buildpacks/heroku-buildpack-php/buildpack.toml')
+        result_packages = list(buildpack.BuildpackHandler.parse(test_file))
+        expected_packages = [
+            models.PackageData(
+                type=buildpack.BuildpackHandler.default_package_type,
+                datasource_id=buildpack.BuildpackHandler.datasource_id,
+                description= "Heroku buildpack for PHP",
+                name="PHP",
+                version=None,
+                extra_data={
+                    "ignore_files": [
+                        ".github/",
+                        ".gitignore",
+                        ".rspec_parallel",
+                        "support/build/",
+                        "support/devcenter/",
+                        "test/",
+                        "Gemfile",
+                        "Gemfile.lock",
+                        "hatchet.json",
+                        "hatchet.lock",
+                        "requirements.txt"
+                    ]
+                }
+            )
+        ]
+        compare_package_results(expected_packages, result_packages)
+
+    def test_parse_paketo_dotnet_execute_buildpack_toml(self):
+        test_file = self.get_test_loc('buildpack/paketo-buildpacks/dotnet-execute/buildpack.toml')
+        result_packages = list(buildpack.BuildpackHandler.parse(test_file))
+        expected_packages = [
+            models.PackageData(
+                type=buildpack.BuildpackHandler.default_package_type,
```

Comment:
YOu need to have these in test expectation files, this would be very hard to add/regenerate on changes otherwise. See the `check_packages_data` function used elsewhere like https://github.com/aboutcode-org/scancode-toolkit/blob/develop/tests/packagedcode/test_cargo.py#L60

Reply:
You don't need to add two tests for these, just checking package data in each manifest case once is fine, and add one full scan (with `run_scan_click` like you've done above) overall for each type of buildpack manifests is enough.